strax/corrections.py

```diff
@@ -75,6 +75,29 @@ def list_corrections(self):
         database = self.client[self.database_name]
         return [x['name'] for x in database.list_collections()]
 
+    def read_at(self, correction, when, limit=1):
+        """Smart logic to read corrections at given time (index), i.e by datetime index
+        :param correction: pandas.DataFrame object name in the DB (str type).
+        :return: DataFrame as read from the corrections database with time
+        index or None if an empty DataFrame is read from the database
+        """
+        before_df = pdm.read_mongo(correction, self.before_date_query(when, limit), self.client[self.database_name])
+        after_df = pdm.read_mongo(correction, self.after_date_query(when, limit), self.client[self.database_name])
+
+        df = pd.concat([before_df, after_df])
```

Comment:
Sorry, but I do not understand this logic here. Why is this different from querying the entire time range at once? Unfortunately, I do not know much about MongoDB queries. Or does this limit mean you pull the last/first entry before and after the specified time?

Reply:
This logic uses the advantages of querying by index, without indexes MongoDB database has to scan every document in a collection to look for the relevant documents. However, using indexes greatly reduces the number of documents MongoDB needs to scan so it is faster